### PR TITLE
Implement Webhook resource

### PR DESCRIPTION
**Pull request by azah**
Add REST routes for webhooks
Add ModifyWebhookPayload and ExecuteWebhookPayload to handle nil
Create webhook struct mapping in separate file
Check for response on execute_webhook if wait is true

### CODE REVIEW

**Comment by z64** [comment]
Would you be interested in implementing the gateway event for webhooks too?

Docs: [`WEBHOOKS_UPDATE`](https://discordapp.com/developers/docs/topics/gateway#webhooks-update)

To be added to [`Client#handle_dispatch`](https://github.com/meew0/discordcr/blob/master/src/discordcr/client.cr#L297)

File: src/discordcr/mappings/rest.cr [hunk 1]
```diff
@@ -15,5 +15,27 @@ module Discord
...
+
+    # A response to the Modify Webhook REST API call.
+    class ModifyWebhookPayload
+      JSON.mapping(name: String?, avatar: String?, channel_id: UInt64?)
+
+      def initialize(@name, @avatar, @channel_id)
+      end
+    end
+
+    class ExecuteWebhookPayload
```
Comment (meew0) [inline src/discordcr/mappings/rest.cr]: Is there any point to having a particular mapping class? It only ever seems to be used once and is never exposed to the user.

File: src/discordcr/mappings/rest.cr [hunk 2]
```diff
@@ -15,5 +15,27 @@ module Discord
         pruned: UInt32
       )
     end
+
+    # A response to the Modify Webhook REST API call.
+    class ModifyWebhookPayload
+      JSON.mapping(name: String?, avatar: String?, channel_id: UInt64?)
```
Comment (meew0) [inline src/discordcr/mappings/rest.cr]: Please make the formatting here consistent with all the other `JSON.mapping` calls (newlines)

File: src/discordcr/mappings/rest.cr [hunk 3]
```diff
@@ -15,5 +15,27 @@ module Discord
         pruned: UInt32
       )
     end
+
+    # A response to the Modify Webhook REST API call.
```
Comment (meew0) [inline src/discordcr/mappings/rest.cr]: Should be "request", not "response".

File: src/discordcr/mappings/rest.cr (hunk 1 above)
Comment (azah) [reply src/discordcr/mappings/rest.cr]: How should it be changed?

**Comment by azah** [comment]
travis: `./deploy.sh: line 15: [: missing `]'`

**Pull request merged**